WordPress/Classes/ViewRelated/Me/App Settings/Boolean User Defaults/BooleanUserDefaultsDebugViewModel.swift

```diff
@@ -0,0 +1,94 @@
+        var filteredSections = BooleanUserDefaultsSections()
+        allUserDefaultsSections.forEach { sectionKey, userDefaults in
+            let filteredUserDefaults = userDefaults.filter { key, userDefault in
+                key.localizedCaseInsensitiveContains(query) || userDefault.title.localizedCaseInsensitiveContains(query)
+            }
+            if sectionKey.localizedCaseInsensitiveContains(query) || !filteredUserDefaults.isEmpty {
+                filteredSections[sectionKey] = filteredUserDefaults.isEmpty ? userDefaults : filteredUserDefaults
+            }
+        }
+        return filteredSections
+    }
+
+    init() {
+        persistentRepository = UserPersistentStoreFactory.instance()
+        coreDataStack = ContextManager.shared
+        load()
+    }
+
+    func load() {
+        let allUserDefaults = persistentRepository.dictionaryRepresentation()
+        var loadedUserDefaultsSections = BooleanUserDefaultsSections()
+
+        for (entryKey, entryValue) in allUserDefaults {
+            if let groupedUserDefaults = entryValue as? [String: Bool] {
+                loadedUserDefaultsSections[entryKey] = processGroupedUserDefaults(groupedUserDefaults)
+            } else if let booleanUserDefault = entryValue as? Bool, !isSystemUserDefault(entryKey) {
+                loadedUserDefaultsSections[Strings.otherBooleanUserDefaultsSectionID, default: [:]][entryKey] = BooleanUserDefault(title: entryKey, value: booleanUserDefault)
+            }
+        }
+
+        allUserDefaultsSections = loadedUserDefaultsSections
+    }
+
+    private func processGroupedUserDefaults(_ userDefaults: [String: Bool]) -> BooleanUserDefaults {
+        userDefaults.reduce(into: BooleanUserDefaults()) { result, keyValue in
+            let (key, value) = keyValue
+            result[key] = processSingleUserDefault(key: key, value: value)
+        }
+    }
+
+    private func processSingleUserDefault(key: String, value: Bool) -> BooleanUserDefault {
+        if let siteID = Int(key), let blogURL = try? Blog.lookup(withID: siteID, in: coreDataStack.mainContext)?.url {
+            return BooleanUserDefault(title: blogURL, value: value)
+        } else {
+            return BooleanUserDefault(title: key, value: value)
+        }
+    }
+
+    func updateUserDefault(_ value: Bool, forSection sectionID: String, forUserDefault userDefaultID: String) {
+        if sectionID == Strings.otherBooleanUserDefaultsSectionID {
+            persistentRepository.set(value, forKey: userDefaultID)
+        } else if var section = allUserDefaultsSections[sectionID] {
+            section[userDefaultID] = BooleanUserDefault(title: userDefaultID, value: value)
+            var sectionValues = section.mapValues { $0.value }
+            persistentRepository.set(sectionValues, forKey: sectionID)
+        }
+        load()
+    }
+
+    private func isSystemUserDefault(_ key: String) -> Bool {
+        key.starts(with: "com.wordpress.")
```

Comment:
[Question] What does a SystemUserDefault mean? And why are we filtering it out?
I removed the filter, and the defaults we're filtering seem to be related to Gutenberg not the system .

Reply:
Good catch. I created the method when I tried to filter out some system flags, but it evolved with no changes in the name. Already fixed that.